### PR TITLE
feature: Integrate Cache utilities with v6.

### PR DESCRIPTION
**Pull request by jimblanc**
#### Description of changes

This change integrates the Cache utilities with the new Singleton and makes them operable with v6. There is no change to the public interface (beyond import path), customers will still be able to:
- Configure the Cache globally
- Create new instances of a Cache with a custom configuration
- Re-configure an existing instance of a Cache

Also made some TypeScript improvements where appropriate.

This most important piece of this PR is the `cacheConfig` function, which contains the logic for resolving the configuration and applying any global or default config values: https://github.com/aws-amplify/amplify-js/pull/11806/files#diff-1d53930c2703b934a6d9cfb22b792f3a82befe22e186250f98f4ac3da7bae5e6R134

#### Description of how you validated changes
Manual testing & unit tests.

#### Checklist

- [x] PR description included
- [x] `yarn test` passes
- [x] Tests are [changed or added](https://github.com/aws-amplify/amplify-js/blob/main/CONTRIBUTING.md#steps-towards-contributions)
- [ ] Relevant documentation is changed or added (and PR referenced)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: packages/core/src/Cache/StorageCache.ts [hunk 1]
```diff
@@ -13,83 +13,66 @@ const logger = new Logger('StorageCache');
...
+
+		this.sanitizeConfig();
 	}
 
 	public getModuleName() {
 		return 'Cache';
 	}
 
-	private checkConfig(): void {
-		// check configuration
```
Comment (jimblanc) [inline packages/core/src/Cache/StorageCache.ts:34]: I felt that some of these basic runtime validations were unnecessary with TypeScript, can add them back if folks think they have value.

Comment (stocaaro) [reply packages/core/src/Cache/StorageCache.ts:34]: Yeah. Seems redundant to ts behavior. Between writing tests to cover this behavior and removing, I would vote for removal for sure.

**Review by stocaaro — DISMISSED** [review]
Seems undifferentiated from other options. I wonder a bit that we're maintaining our own LRU variant. Do our customers use this? Does our internal use justify its existence?

**Review by stocaaro — APPROVED** [review]

**Review by israx — APPROVED** [review]
LGTM

File: packages/core/src/Cache/Utils/CacheUtils.ts [hunk 2]
```diff
@@ -88,3 +88,5 @@ export class CacheObject {
 		delete store[key];
 	}
 }
+
+export const getCurrSizeKey = (keyPrefix: string) => keyPrefix + 'CurSize';
```
Comment (cshfang) [inline packages/core/src/Cache/Utils/CacheUtils.ts:92]: Nit: Think this may have been a typo: Cur

May prefer string interp

**Pull request merged**